src/core/namespace.c

```diff
@@ -1073,6 +1074,19 @@ static int mount_sysfs(const MountEntry *m) {
         return 1;
 }
 
+static bool mount_option_supported(const char *fstype, const char *key, const char *value) {
+        _cleanup_close_ int fd = -1;
+
+        /* This function assumes support by default. Only if the fsconfig() call fails with -EINVAL/-EOPNOTSUPP
+         * will it report that the option/value is not supported. */
+
+        fd = fsopen(fstype, FSOPEN_CLOEXEC);
+        if (fd < 0)
+                return true; /* If fsopen() fails for whatever reason, assume the value is supported. */
+
+        return !IN_SET(RET_NERRNO(fsconfig(fd, FSCONFIG_SET_STRING, key, value, 0)), -EINVAL, -EOPNOTSUPP);
```

Comment:
similar here

Reply:
ERRNO_IS_NOT_SUPPORTED()?

i mean, ENOSYS is what we have to expect here, no?

Reply:
and fsconfig() doesn't warn on old kernels?

also, isn't fsconfig() so recent that the same kernels which support it also support the new hidepid= funcitonality?

Reply:
or to turn this around: on old kernels, where people are annoyed about the log message, this all is not going to work, is it?

Reply:
Redhat has backported fsopen()/fsconfig() to RHEL 8 and centos stream 8. They actually suggested using this approach. fsopen()/fsconfig() are from 5.2, hidepid is from 5.8, so with that and the fact it got backported, this should generally improve the situation a bit.

Reply:
can you add a comment about all this to the sources? i.e. a discussion why to test this explicitly, and when fsopen() and the new hidepid= was added upstream (don't mention which rhel versions backported it, just say that some distros did)